js/views/reader_view.js

```diff
@@ -547,11 +547,7 @@ ReadiumSDK.Views.ReaderView = function(options) {
 
                 var spineItem = _spine.getItemById(bookMark.idref);
                 
-                initViewForItem(spineItem, function(isViewChanged){
-
-                    if(!isViewChanged) {
-                        _currentView.setViewSettings(_viewerSettings);
-                    }
+                initViewForItem(spineItem, function(isViewChanged) {
```

Comment:
Not directly related to your code changes, but I've just noticed: isn't this event fired twice?
`self.trigger(ReadiumSDK.Events.SETTINGS_APPLIED);`
(the return statement is in a function callback, not in the main function, so the `if` condition cascades down into a redundant and early `trigger()` call)